pkg/tracker/dependency/tracker.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2021 VMware
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
```

Comment:
should we include a reference to the original implementation? sounds like this is heavily based on https://github.com/vmware-labs/reconciler-runtime/blob/29a418a3795edc5b870e69f28e1b611a33dcad3b/tracker/tracker.go (in that case, I believe we should have attribution to `knative` as it's apache v2)

Reply:
fixed all this in latest commit, thanks!